nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/AdminAdapter.java

```diff
@@ -598,18 +599,9 @@ void decodePasswords(ParameterMap pmap, final String requestString) {
             }
             String paramName = token.substring(0, token.indexOf("="));
             String value = token.substring(token.indexOf("=") + 1);
-
-            try {
-                value = URLDecoder.decode(value, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                aalogger.log(Level.WARNING, KernelLoggerInfo.cantDecodeParameter,
-                        new Object[] { paramName, value });
-                continue;
-            }
-
             try {
-                value = new String(decoder.decodeBuffer(value));
-            } catch (IOException e) {
+                value = new String(Base64.getUrlDecoder().decode(value), UTF_8);
```

Comment:
And here?

Reply:
Here it should be ok too - it is used as a part of URL (query string). We should remove these parts as sending passwords in URL is really awful security bug - but that is for another PR.